src/containers/StructurePage/plannedResource/AddExistingResource.tsx

```diff
@@ -158,45 +139,101 @@ const AddExistingResource = ({ onClose, resourceTypes, existingResourceIds, node
     }
   };
 
-  const onPaste = async (evt: ChangeEvent<HTMLInputElement>) => {
-    setError('');
-    setContent(undefined);
-    setPastedUrl(evt.currentTarget.value);
+  const onAddResource = async () => {
+    if (!preview) return;
 
-    const input = evt.target.value;
+    let id = resourceId;
+
+    if (!resourceId) {
+      const isLearningpath = selectedType === RESOURCE_TYPE_LEARNING_PATH && 'metaUrl' in preview;
+      const isArticleOrDraft = 'paths' in preview;
+      id = isLearningpath
+        ? await findResourceIdLearningPath(preview.id)
+        : isArticleOrDraft
+        ? getResourceIdFromPath(preview.paths?.[0])
+        : undefined;
+    }
 
-    if (!input) {
+    if (existingResourceIds.includes(String(resourceId))) {
+      resetPastedUrlStatesWithError(t('taxonomy.resource.addResourceConflict'));
       return;
     }
 
-    const isNumber = /^-?\d+$/.test(input);
-    const articleInPathMatch = input.match(/article\/(\d+)/);
-    const articleId = articleInPathMatch ? articleInPathMatch[1] : isNumber && input;
-    const resourceId = getResourceIdFromPath(evt.target.value);
+    if (!id) return;
+
+    await createNodeResource({ body: { resourceId: id, nodeId }, taxonomyVersion })
+      .then((_) => onClose())
+      .catch(() => resetPastedUrlStatesWithError('taxonomy.resource.creationFailed'));
+    setLoading(false);
+  };
+
+  const resetPastedUrlStatesWithError = (error?: string) => {
+    error && setError(error);
+    setPreview(undefined);
+    setPreviewLoading(false);
+    setResourceId(undefined);
+  };
+
+  const onPaste = async (evt: ChangeEvent<HTMLInputElement>) => {
+    resetPastedUrlStatesWithError('');
+    const input = evt.target.value;
+    setPastedUrl(input);
+    if (!input) return;
+
+    const pastedIsNumber = /^-?\d+$/.test(input);
+    const articleIdInPathMatch = input.match(/article\/(\d+)/);
+    const articleIdInInput = articleIdInPathMatch
+      ? articleIdInPathMatch[1]
+      : pastedIsNumber && input;
+
+    let resourceId = undefined;
+    let preview = undefined;
+    setPreviewLoading(true);
 
     try {
-      let id = Number(articleId);
-      setPreviewLoading(true);
+      if (articleIdInInput) {
```

Comment:
En anna ting. Her har du et nummer og henter først artikkelen, så bruker du artikkel.id for å søke. article brukes ikkje til nokke. Kunne du ikkje berre brukt nummeret direkte mot søket? Om du får tomt resultat (gitt at du bruker search og ikkje searchResources) betyr at artikkelen ikkje finnes.
Og om poenget er å finne resourceId er det bedre å bruke useNodes og sende inn `contentURI: urn:article:articleId` som parameter. Då sjekker du strukturen direkte.